charts/gateway-operator/templates/deployment.yaml

```
@@ -0,0 +1,83 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: {{ template "kong.fullname" . }}-controller-manager
+  namespace:  {{ template "kong.namespace" . }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        control-plane: controller-manager
+    spec:
+      containers:
+      - args:
+        - --health-probe-bind-address=:8081
+        - --metrics-bind-address=127.0.0.1:8080
+        command:
+        - /manager
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+          initialDelaySeconds: 15
+          periodSeconds: 20
+        name: manager
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8081
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        resources:
+          limits:
+            cpu: 500m
+            memory: 256Mi
+          requests:
+            cpu: 10m
+            memory: 128Mi
+        securityContext:
+          allowPrivilegeEscalation: false
```

This should match what we use for the proxy/controller to allow the operator to run in Restricted SCC namespaces:

```suggestion
          allowPrivilegeEscalation: false
          readOnlyRootFilesystem: true
          runAsNonRoot: true
          seccompProfile:
            type: RuntimeDefault
          capabilities:
            drop:
            - ALL
```

Hopefully I didn't break the spacing in the suggestion.

Minor note that we should not need the `runAsUser` here, since the operator uses a numeric UID in its Dockerfile.

---

Great! Thank you!

I would like to provide this as a follow-up PR, and I have also tracked it in https://github.com/Kong/gateway-operator/issues/1323

---

To clarify, this doesn't need to be configurable, we should just use the static value above unless we find some reason it actually needs to be configurable.

The `kong` chart does let you set it, but AFAIK there's no real reason to. There are some debugging scenarios where you need to add capabilities, but for such debugging tasks I think we should just ask users to modify the Deployment temporarily.

There are some weird caveats around OpenShift if you have `runAsUser`, but this doesn't, and there was maybe a reason to let you modify it when PodSecurityPolicy was still standard. https://kubernetes.io/docs/concepts/security/pod-security-standards/ doesn't have an option for custom policies that I know of.

---

OK, make sense